core/src/main/java/io/grpc/ManagedChannelBuilder.java

```diff
@@ -118,8 +118,8 @@
   /**
    * Provides a custom {@code User-Agent} for the application.
    *
-   * <p>It's an optional parameter. If provided, the given agent will be prepended by the
-   * grpc {@code User-Agent}.
+   * <p>It's an optional parameter. The library will provide a user agent independent of this
+   * option. If provided, the given agent will appear before the library's user agent information.
```

Comment:
s/appear before/prepend ? Otherwise its unclear where it will appear. Http2Headers?